### PR TITLE
Better extraction for nrk

### PR DESCRIPTION
**Pull request by Roxedus**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

This fixes two open issues for the NRK extractor, #25594 and #24221 as well as providing `alt_title`, useful for shows that does indeed have unique episode names, rather than `<series name> <episode number>:<total episodes in season>`

In order to fully fix 24221 I had to use a additional endpoint to fetch a m3u8 playlist, this seems to be the endpoint which exposes 1080P quality.

Some test maintenance was also done, the description seems to change from time to time.

### CODE REVIEW

File: youtube_dl/extractor/nrk.py [hunk 1]
```diff
@@ -52,6 +57,28 @@ def _real_extract(self, url):
         def make_title(t):
             return self._live_title(t) if live else t
 
+        playback_convia = playback_manifest.get('statistics').get('conviva')
```
Comment (dstftw) [inline youtube_dl/extractor/nrk.py:60]: Breaks if request fails. Read coding conventions on optional metadata.

File: youtube_dl/extractor/nrk.py [hunk 2]
```diff
@@ -52,6 +57,28 @@ def _real_extract(self, url):
...
+        if playback_convia:
+            streamurl = playback_convia.get('streamUrl', None)
+            stream = self._extract_m3u8_formats(streamurl, video_id, 'mp4',
+                'm3u8_native', m3u8_id='hls', fatal=False)
+            custom = playback_convia.get('custom')
+
+            dur = parse_duration(playback_convia.get('duration'))
+            subs = {}
+            for sub_title in playback_manifest.get('playable').get('subtitles'):
```
Comment (dstftw) [inline youtube_dl/extractor/nrk.py:69]: Ditto.

File: youtube_dl/extractor/nrk.py [hunk 3]
```diff
@@ -139,6 +166,13 @@ def video_id_and_title(idx):
                 EPISODENUM_RE, _season_episode, 'episode number',
                 default=None, group='episode'))
 
+            if not season_number or episode_number:
+                programs = self._download_json(
+                    'http://%s/programs/%s' % (self._api_host, video_id),
+                    video_id, 'Downloading programs manifest JSON', fatal=False)
+                season_number = int_or_none(programs.get('seasonNumber'))
+                episode_number = int_or_none(programs.get('episodeNumber'))
```
Comment (dstftw) [inline youtube_dl/extractor/nrk.py:174]: Ditto.

File: youtube_dl/extractor/nrk.py [hunk 4]
```diff
@@ -427,7 +464,8 @@ def _real_extract(self, url):
 
         nrk_id = self._parse_json(
             self._search_regex(JSON_LD_RE, webpage, 'JSON-LD', group='json_ld'),
-            display_id)['@id']
+            display_id).get('@id') or \
```
Comment (dstftw) [inline youtube_dl/extractor/nrk.py:467]: Breaks.

**Comment by Roxedus** [comment]
> making them appear as playlists
They always were marked as playlist, the "duplication" is probably because it pulls from two sources

File: youtube_dl/extractor/nrk.py (hunk 4 above)
Comment (Roxedus) [reply youtube_dl/extractor/nrk.py:468]: Because of the lack of fallback?

**Pull request closed without merging**